pyamg/relaxation/smoothing.py

```diff
@@ -136,37 +138,177 @@ def change_smoothers(ml, presmoother, postsmoother):
     elif not isinstance(postsmoother, list):
         raise ValueError('Unrecognized postsmoother')
 
-    # set ml.levels[i].presmoother = presmoother[i]
-    i = 0
-    for i in range(min(len(presmoother), len(ml.levels[:-1]))):
+
+    # set ml.levels[i].presmoother = presmoother[i],
+    #     ml.levels[i].postsmoother = postsmoother[i]
+    fn1 = None      # Predefine to keep scope beyond first loop
+    fn2 = None
+    kwargs1 = {}
+    kwargs2 = {}
+    min_len = min(len(presmoother), len(postsmoother), len(ml.levels[:-1]))
+    same = (len(presmoother) == len(postsmoother))
+    for i in range(0, min_len):
         # unpack presmoother[i]
-        fn, kwargs = unpack_arg(presmoother[i])
+        fn1, kwargs1 = unpack_arg(presmoother[i])
         # get function handle
         try:
-            setup_presmoother = eval('setup_' + str(fn))
+            setup_presmoother = eval('setup_' + str(fn1))
         except NameError:
-            raise NameError("invalid presmoother method: ", fn)
-        ml.levels[i].presmoother = setup_presmoother(ml.levels[i], **kwargs)
-
-    # Fill in remaining levels
-    for j in range(i+1, len(ml.levels[:-1])):
-        ml.levels[j].presmoother = setup_presmoother(ml.levels[j], **kwargs)
+            raise NameError("invalid presmoother method: ", fn1)
+        ml.levels[i].presmoother = setup_presmoother(ml.levels[i], **kwargs1)
 
-    # set ml.levels[i].postsmoother = postsmoother[i]
-    i = 0
-    for i in range(min(len(postsmoother), len(ml.levels[:-1]))):
         # unpack postsmoother[i]
-        fn, kwargs = unpack_arg(postsmoother[i])
+        fn2, kwargs2 = unpack_arg(postsmoother[i])
         # get function handle
         try:
-            setup_postsmoother = eval('setup_' + str(fn))
+            setup_postsmoother = eval('setup_' + str(fn2))
         except NameError:
-            raise NameError("invalid postsmoother method: ", fn)
-        ml.levels[i].postsmoother = setup_postsmoother(ml.levels[i], **kwargs)
+            raise NameError("invalid postsmoother method: ", fn2)
+        ml.levels[i].postsmoother = setup_postsmoother(ml.levels[i], **kwargs2)
+
+        # Check if symmetric smoothing scheme
+        try:
+            it1 = kwargs1['iterations']
+        except:
+            it1 = 1
+        try:
+            it2 = kwargs2['iterations']
+        except:
+            it2 = 1
+        if (fn1 != fn2) or (it1 != it2):
+            ml.symmetric_smoothing = False
+        elif (fn1 != 'jacobi') and (fn1 != 'richardson') and \
```

Comment:
I think that you can you define a variable at the top of the file, like

------------
A = 5
...
def temp(x)
    print x + A

----------------

Also, I just remembered, convention usually puts constant default values in all caps, so maybe we want 
DEFAULT_SWEEP_DIRECTION
DEFAULT_NUM_ITERS
MAT_VEC_STYLE_RELAXATION
or 
DEFAULT_SWEEP
DEFAULT_NITER
MAT_VEC_STYLE_RELAXATION